client/src/API.ts

```diff
@@ -7,76 +7,109 @@ import {
 } from "./Command";
 import { TextStyles } from "./components/Typewriter";
 
-interface SGProps {
-  handleConversation: Function;
+const BASE_URL = "http://127.0.0.1:5000/";
+
+export interface SGProps {
+  handleConversation: (text: string[], styles: TextStyles[]) => void;
   setGameID: Dispatch<SetStateAction<string>>;
-  isMounted: () => boolean;
+  handleUserInput: (input: string) => void;
 }
 
-export const startGame = async ({
-  handleConversation,
-  setGameID,
-  isMounted,
-}: SGProps) => {
-  const response = await fetch("http://127.0.0.1:5000/start");
+const fetchGame = async (
+  endpoint: string,
+  gameID: string,
+  userIn: string = ""
+) => {
+  return fetch(`${BASE_URL}${endpoint}${gameID}`, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({ input: userIn }),
+  });
+};
 
-  if (!response.ok || !isMounted()) {
-    return;
+const getGameID = (): string => localStorage.getItem("game_id") as string;
```

Comment:
This `"game_id"` string you use in multiple places could/should be a `const`.

Reply:
I mean add `const GAME_ID_KEY: string = "game_id"` to the top of the file and use `GAME_ID_KEY` here in lieu of the hardcoded string. This could help us avoid bugs later if we ever decide to change the name of the key.